src/session_impl.cpp

```diff
@@ -6865,6 +6950,7 @@ namespace aux {
 			, struct tracker_response const& resp)
 		{
 			TORRENT_UNUSED(tracker_ips);
+			if (!should_log()) return;
```

Comment:
take a look at where the tracker_logger is created. If logging is disabled, we shouldn't create it in the first place